test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -1632,6 +1633,7 @@ protected Settings nodeSettings(int nodeOrdinal) {
         Settings.Builder builder = Settings.builder()
                 // Default the watermarks to absurdly low to prevent the tests
                 // from failing on nodes without enough disk space
+                .put(NodeEnvironment.MAX_LOCAL_STORAGE_NODES_SETTING.getKey(), 8)
```

Comment:
I feel like we should either use a guess from the annotation or just go with Integer.MAX_VALUE. If we do stick with 8 we should add a comment about why 8.

Also, the comment above this line is actually for the next line. Maybe just shift this line above the comment?